src/modules/credentials/CredentialUtils.ts

```diff
@@ -1,10 +1,40 @@
-import type { CredentialPreviewAttribute } from './messages/CredentialPreview'
+import type { LinkedAttachment } from '../../utils/LinkedAttachment'
 import type { CredValues } from 'indy-sdk'
 
 import BigNumber from 'bn.js'
 import { sha256 } from 'js-sha256'
 
+import { AriesFrameworkError } from '../../error'
+import { HashlinkEncoder } from '../../utils/HashlinkEncoder'
+
+import { CredentialPreview, CredentialPreviewAttribute } from './messages/CredentialPreview'
+
 export class CredentialUtils {
+  /**
+   * Adds a credential to the preview where it links to the attachment
+   *
+   * @param attachments a list of the attachments that need to be linked to a credential
+   * @param preview the credential previews where the new linked credential has to be appended to
+   *
+   * @returns a modified version of the credential preview with the linked credentials
+   * */
+  public static createAndLinkCredentialToAttachment(attachments: LinkedAttachment[], preview: CredentialPreview) {
```

Comment:
Maybe?

```suggestion
  public static createAndLinkAttachmentsToPreview(attachments: LinkedAttachment[], preview: CredentialPreview) {
```


Also the meaning of the following sentence is not fully clear to me. I wouldn't use credentials here, but rather attachments and preview

```
Adds a credential to the preview where it links to the attachment
```